app/queries/admin/users_query.rb

```diff
@@ -46,16 +48,24 @@ def self.filter_organization_memberships(relation:, organizations:)
     #       query per role is inadequate.
     #
     # @see https://github.com/RolifyCommunity/rolify/blob/0c883f4173f409766338b9c6dfc64b0fc8ec8a52/lib/rolify/finders.rb#L26-L32
-    def self.filter_roles(relation:, roles:, role_map: Constants::Role::SPECIAL_ROLES_LABELS_TO_WHERE_CLAUSE)
+    def self.filter_roles(relation:, roles:, role_map: Constants::Role::ALL_ROLES_LABELS_TO_WHERE_CLAUSE)
       conditions = []
       values = []
 
       # Assemble the conditions and positional parameter values
       roles.each do |role|
         where_clause = role_map.fetch(role)
         resource_type = where_clause.fetch(:resource_type, nil)
+        name = where_clause.fetch(:name)
+
+        if name.nil?
+          # TODO: When we filter by "Good standing" status, we want to filter by
+          # NOT any of [trusted, warned, suspended, comment_suspended]
+          next
+        end
```

Comment:
At the moment I'm skipping "Good standing" 😅 

I had a bash at the query for this but quickly realised I'm going to need assistance. I believe what we want is:

- Users who have NO roles
- Users who have a role, but NOT any of `[trusted, warned, suspended, comment_suspended]`